packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -64,11 +64,18 @@ void ensureDirectoryExists(String filePath) {
   }
 }
 
-/// Recursively copies `srcDir` to `destDir`, invoking [onFileCopied] if
-/// specified for each source/destination file pair.
+// Creates `destDir` if needed, then recursively copies `srcDir` to `destDir`,
+/// invoking [onFileCopied], if specified, for each source/destination file pair.
 ///
-/// Creates `destDir` if needed.
-void copyDirectorySync(Directory srcDir, Directory destDir, [ void onFileCopied(File srcFile, File destFile) ]) {
+/// Skips files if [shouldCopyFile] returns `false`.
+void copyDirectorySync(
+  Directory srcDir,
+  Directory destDir,
+  {
+    bool shouldCopyFile(File srcFile, File destFile),
```

Comment:
rather than a callback closure, what if this instead was a optional boolean argument `shouldOverwrite` that defaulted to `true`?

Reply:
and then you implement the logic in the body of the `copyDirectorySync()`

Reply:
I thought about it. I ended up going with a callback because this way I see this as a generic utility--someone may want to filter based on other criteria.

Reply:
:+1: